lib/src/builder/codegen/typed_map_impl_generator.dart

```diff
@@ -46,7 +46,7 @@ abstract class TypedMapImplGenerator extends BoilerplateDeclarationGenerator {
 
   TypedMapNames get names;
   bool get isComponent2;
-  FactoryNames get factoryNames;
+  List<FactoryNames> get factoryNames;
```

Comment:
We need to generate a props config for every function component factory (even if they share the same props mixin) so that the display name is set up correctly. This has to be a list in order to get all the factory names from `PropsMapViewOrFunctionComponentDeclaration.factories`. In all other cases `factoryNames` would have a single element -- I just added an assert for that in `_generateFactory`.